ocean_lib/common/ddo/ddo.py

```diff
@@ -282,73 +288,77 @@ def get_service_by_index(self, index):
                 return service
 
         # try to find by type
+        # FIXME: this shouldn't work and is possibly a typo
```

Comment:
Have you created an issue for this?

Reply:
I did now :) https://github.com/oceanprotocol/ocean.py/issues/407